pkg/kubelet/kubelet.go

```diff
@@ -3282,7 +3311,21 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
 
 func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.PodStatus) api.PodStatus {
 	glog.V(3).Infof("Generating status for %q", format.Pod(pod))
+
+	// any pod that requires eviction must be added to the list to sync.
+	attrs := &lifecycle.PodEvictAttributes{Pod: pod}
+	for _, podEvictor := range kl.podEvictors {
+		if result := podEvictor.Evict(attrs); result.Evict {
```

Comment:
@derekwaynecarr By the time we get here, the eviction has already taken place right?
Is that the motivation for the logic [above](https://github.com/kubernetes/kubernetes/pull/24344/files#diff-bf28da68f62a8df6e99e447c4351122dR2056) as well?